ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -141,8 +142,18 @@ boolean scheduleNextForCompaction(BlobStore store) {
   }
 
   /**
+   * Disable the given {@code store} for compaction.
+   * @param store the {@link BlobStore} to be disabled.
+   * @return {@code true} if the disabling was successful. {@code false} if not.
+   */
+  boolean disableCompactionForBlobStore(BlobStore store) {
+    if (compactionExecutor == null) {
+      return true;
+    }
+    return compactionExecutor.disableCompactionForBlobStore(store);
```

Comment:
minor: you can simplify this by
`return compactionExecutor == null || compactionExecutor.disableCompactionForBlobStore(store)`